src/els_completion_provider.erl

```diff
@@ -232,50 +268,23 @@ record_fields(Document, RecordName) ->
 
 -spec find_record_definition(els_dt_document:item(), atom()) -> [poi()].
 find_record_definition(Document, RecordName) ->
-  POIs = lists:flatten([ local_definitions(Document, record)
-                       , included_definitions(Document, record)
-                       ]),
+  POIs = local_and_included_pois(Document, record),
   [X || X = #{id := Name} <- POIs, Name =:= RecordName].
 
 %%==============================================================================
 %% Macros and Records
 %%==============================================================================
 
--type def_type() :: define | record.
-
--spec definitions(els_dt_document:item(), def_type()) -> [map()].
+-spec definitions(els_dt_document:item(), poi_kind()) -> [map()].
 definitions(Document, Type) ->
-  POIs = lists:flatten([ local_definitions(Document, Type)
-                       , included_definitions(Document, Type)
-                       ]),
+  POIs = local_and_included_pois(Document, Type),
   Defs = [ #{ label => atom_to_binary(Name, utf8)
             , kind  => completion_item_kind(Type)
             }
            || #{id := Name} <- POIs
          ],
   lists:usort(Defs).
 
--spec local_definitions(els_dt_document:item(), def_type()) -> [map()].
```

Comment:
These functions were generic enough that I could just use them without any modification to their logic so I moved them to the general `Helper Functions` sections.